internal/inventory/config.go

```diff
@@ -30,48 +26,52 @@ type ApplicationSettings struct {
 }
 
-// WriteableConfig is a struct representation of the Writeable section of the configuration.toml file.
-type WriteableConfig struct {
-	LogLevel string
+// CustomConfig is the struct representation of all of the sections from the configuration.toml
+// that we are interested in syncing with Consul.
+type CustomConfig struct {
+	AppSettings ApplicationSettings
+	Aliases     map[string]string
 }
 
-// ConsulConfig is the struct representation of all of the sections from the configuration.toml
-// that we are interested in syncing with Consul.
-type ConsulConfig struct {
-	Writable            WriteableConfig
-	ApplicationSettings ApplicationSettings
-	Aliases             map[string]string
+type ServiceConfig struct {
+	AppCustom CustomConfig
+}
+
+// UpdateFromRaw updates the service's full configuration from raw data received from
+// the Service Provider.This function implements the UpdatableConfig interface for ServiceConfig.
+func (c *ServiceConfig) UpdateFromRaw(rawConfig interface{}) bool {
+	configuration, ok := rawConfig.(*ServiceConfig)
+	if !ok {
+		return false
+	}
+
+	*c = *configuration
+
+	return true
 }
 
 var (
-	// ErrUnexpectedConfigItems is returned when the input configuration map has extra keys
-	// and values that are left over after parsing is complete
-	ErrUnexpectedConfigItems = errors.New("unexpected config items")
-	// ErrMissingRequiredKey is returned when we are unable to parse the value for a config key
-	ErrMissingRequiredKey = errors.New("missing required key")
 	// ErrOutOfRange is returned if a config value is syntactically valid for its type,
 	// but otherwise outside of the acceptable range of valid values.
 	ErrOutOfRange = errors.New("config value out of range")
 )
 
-// NewConsulConfig returns a new ConsulConfig instance with default values.
-func NewConsulConfig() ConsulConfig {
-	return ConsulConfig{
-		Aliases: map[string]string{},
-		Writable: WriteableConfig{
-			LogLevel: "INFO",
-		},
-		ApplicationSettings: ApplicationSettings{
-			MobilityProfileThreshold:     6,
-			MobilityProfileHoldoffMillis: 500,
-			MobilityProfileSlope:         -0.008,
-			DeviceServiceName:            "edgex-device-llrp",
-			DeviceServiceURL:             "http://edgex-device-llrp:49989/",
-			MetadataServiceURL:           "http://edgex-core-metadata:48081/",
-			DepartedThresholdSeconds:     600,
-			DepartedCheckIntervalSeconds: 30,
-			AgeOutHours:                  336,
-			AdjustLastReadOnByOrigin:     true,
+// NewServiceConfig returns a new ServiceConfig instance with default values.
+func NewServiceConfig() ServiceConfig {
+	return ServiceConfig{
+		AppCustom: CustomConfig{
+			Aliases: map[string]string{},
+			AppSettings: ApplicationSettings{
+				MobilityProfileThreshold:     6,
+				MobilityProfileHoldoffMillis: 500,
+				MobilityProfileSlope:         -0.008,
+				DeviceServiceName:            "edgex-device-llrp",
+				DeviceServiceURL:             "http://edgex-device-llrp:49989/",
```

Comment:
I'm actually not sure what to put for the Device service url, but i think the ports changed at least.

```go
				DeviceServiceName:            "device-rfid-llrp",
				DeviceServiceURL:             "http://localhost:59989/",
```

Reply:
Goes away once we switch to the Command Client

Reply:
switched to command client and fixed DeviceServiceName and removed DeviceServiceUrl